zipline/data/us_equity_minutes.py

```diff
@@ -11,27 +11,402 @@
+    e.g.:
+    2016-01-19 14:31
+    2016-01-19 14:32
+    ...
+    2016-01-19 20:59
+    2016-01-19 21:00
+    2016-01-20 14:31
+    2016-01-20 14:32
+    ...
+    2016-01-20 20:59
+    2016-01-20 21:00
+
+    All assets are written with a common 'index', sharing a common first
+    trading day. Assets that do not begin trading until after the first trading
+    day will have zeros for all pricing data up and until data is traded.
+
+    'index' is in quotations, because bcolz does not provide an index. The
+    format allows index-like behavior by writing each minute's data into the
+    corresponding position of the enumeration of the aforementioned datetime
+    index.
+
+    The datetimes which correspond to each position are written in the metadata
+    as integer nanoseconds since the epoch into the `minute_index` key.
+    """
+    def __init__(self,
+                 first_trading_day,
+                 rootdir,
+                 market_opens,
+                 minutes_per_day=MINUTES_PER_DAY,
+                 ohlc_ratio=OHLC_RATIO,
+                 expectedlen=DEFAULT_EXPECTEDLEN):
+        """
+        Parameters:
+        -----------
+
+        first_trading_day : datetime-like
+            The first trading day in the data set.
+        
+        rootdir : string
+            Path to the root directory into which to write the metadata and
+            bcolz subdirectories.
+
+        market_opens : pd.Series
+            The market opens used as a starting point for each periodic span of
+            minutes in the index.
+
+        minutes_per_day : int
+            The number of minutes per each period. Defaults to 390, the mode
+            of minutes in NYSE trading days.
+
+        ohlc_ratio : int
+            The ratio by which to multiply the pricing data to convert the floats
+            from floats to an integer to fit within the np.uint32.
+
+            The default is 1000 to support pricing data which comes in to the
+            thousands place.
+
+        expectedlen : int
+            The expected length of the dataset, used when creating the initial
+            bcolz ctable.
+
+            If the expectedlen is not used, the chunksize and corresponding
+            compression ratios are not ideal.
+
+            Defaults to supporting 15 years of market data.
+
+            see: http://bcolz.blosc.org/opt-tips.html#informing-about-the-length-of-your-carrays
+        """
+        self._rootdir = rootdir
+        self._first_trading_day = first_trading_day
+        self._market_opens = market_opens[
+            market_opens.index.slice_indexer(start=self._first_trading_day)]
+        self._trading_days = market_opens.index
+        self._minutes_per_day = minutes_per_day
+        self._expectedlen = expectedlen
+        self._ohlc_ratio = ohlc_ratio
+
+        self._minute_index = _calc_minute_index(
+            self._market_opens, self._minutes_per_day)
+
+        self._write_metadata()
```

Comment:
Since the `write` is being called multiple times, I think writing the metadata eagerly works better.
However, admittedly it does blur the dataset directory's initial creation and successive updates.

It should have no functional difference to pass an ever expanding list of market_opens, however passing a list that is different otherwise would certainly cause issues.

This is a piece that could be made more robust, by checking the length and values of the `minute_index` when creating a writer for a pre-existing directory and failing if the values differ, but way be able to defer adding that robustness.

I will attempt adding at least a basic check now.